src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -143,4 +264,44 @@ public AsyncJsonTokenParser(
     return Futures.transformAsync(futureVerifiers, checkNullFunction, executor);
   }
 
+  /**
+   * Converts exceptions (when applicable) to {@link InvalidJsonTokenException}
+   * for better exception handling in the asynchronous parser.
+   */
+  private Exception getRethrownException(Exception originalException) {
```

Comment:
nit: Let's call the method mapException() as it's unclear what rethrown exception means (the exception hasn't been rethrown yet)

Reply:
Changed.